src/jail_rules.c

```diff
@@ -908,8 +916,31 @@ jail_rules_new(
     char** section,
     GError** error)
 {
-    return jail_rules_from_data(jail_rules_build(program, conf, opt,
-        profile_path, section, error));
+    JailRules* rules = jail_rules_from_data(jail_rules_build(program,
+        conf, opt, profile_path, section, error));
+    if (error && *error && opt->profile && conf->desktop_dir &&
+            g_str_has_suffix(opt->profile, SAILJAIL_PROFILE_SUFFIX) &&
+            (g_error_matches(*error, G_FILE_ERROR, G_FILE_ERROR_NOENT) ||
+             g_error_matches(*error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_PARSE))
+            ) {
+        /* User supplied .profile file which wasn't appropriate */
+        char* name = g_path_get_basename(opt->profile);
+        name[strlen(name)-sizeof(SAILJAIL_PROFILE_SUFFIX)+1] = '\0';
```

Comment:
I wonder if it would be better to use `strrchr(name, '.')` here instead. Technically it doesn't matter that much since we checked that there is that suffix which also begins with a dot but I guess it might be more readable. That would require more lines though. :grin:

Reply:
Performance wise it is hard to beat that strchr() way, but IMO optimizing this to max makes no sense. Both variants look like they have potential to crash, and reader must inspect the surrounding code to gauge whether such outcomes are actually possible or not... How about:

    name[strcspn(name, ".")] = 0;

Reply:
Oh, that so nice. :+1: Changed it.

Reply:
Except... this one matches dots from the beginning of the string and if there are more than one dot, then this will not work correctly. It's quite plausible that in the future the names may be in a format that contains more than one dot.